### PR TITLE
docs(user): move examples and recipes to Python files

### LINKED ISSUE #44: Add example code & recipes to test suite
Need to do this in a DRY manner.

### PR DESCRIPTION
**Pull request by vytas7**
Fixes #44.

### CODE REVIEW

File: examples/recipes/output_csv_stream_asgi.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        writer.writerow(('n', 'Fibonacci Fn'))
+
+        previous = 1
+        current = 0
+        for i in range(n + 1):
+            writer.writerow((i, current))
+            previous, current = current, current + previous
+
+            for chunk in stream.result:
+                yield chunk
```
Comment (CaselIT) [inline examples/recipes/output_csv_stream_asgi.py:27]: yield from? this seems the only difference from the wsgi one (other than the async bits)

File: examples/recipes/request_id_context.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# context.py
+
+import threading
```
Comment (CaselIT) [inline examples/recipes/request_id_context.py:3]: can't you just update this here, instead of having another PR for #2260 ?

Comment (CaselIT) [reply examples/recipes/request_id_context.py:3]: I don't think you would even need this wrapper class.

File: tests/conftest.py [hunk 3]
```diff
@@ -17,6 +24,78 @@ def asgi(request):
     return request.param
 
 
+class _SuiteUtils:
```
Comment (CaselIT) [inline tests/conftest.py:27]: you didn't remove the util file

File: tests/test_examples.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    )
+
+
+def test_things_advanced(asgi, util):
+    suffix = '_asgi' if asgi else ''
+    advanced = util.load_module(f'examples/things_advanced{suffix}.py')
+
+    # NOTE(vytas): The ASGI example explicitly requires Content-Length
+    #   (its middleware errors out otherwise with 400).
+    #   Should we change this?
```
Comment (CaselIT) [inline tests/test_examples.py:23]: makes sense.

File: tests/test_recipes.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        'recipe,expected_head',
+        [
+            ('output_csv_text', '"fruit","quantity"\r\n"apples",13\r\n'),
+            ('output_csv_stream', '"n","Fibonacci Fn"\r\n0,0\r\n1,1\r\n'),
+        ],
+        ids=['simple', 'stream'],
+    )
+    def test_csv_output(self, asgi, util, recipe, expected_head):
+        suffix = 'asgi' if asgi else 'wsgi'
+        module = util.load_module(f'examples/recipes/{recipe}_{suffix}.py')
```
Comment (CaselIT) [inline tests/test_recipes.py:65]: maybe it would make sense to have a `util.load_recepie(recepie, asgi)` since a similar thing is used multiple times

**Review by CaselIT — COMMENTED** [review]
nice work!

File: examples/recipes/request_id_context.py (hunk 2 above)
Comment (vytas7) [reply examples/recipes/request_id_context.py:3]: I didn't want to alter that recipe too much, I'm not even the author. IIRC it was created by various people during PyCon US 2019 Sprints.

File: examples/recipes/output_csv_stream_asgi.py (hunk 1 above)
Comment (vytas7) [reply examples/recipes/output_csv_stream_asgi.py:27]: See [the note on the recipe itself](https://falcon.readthedocs.io/en/latest/user/recipes/output-csv.html#streaming-large-csv-files-on-the-fly); also here: [Asynchronous `yield from`](https://peps.python.org/pep-0525/#asynchronous-yield-from).

File: tests/conftest.py (hunk 3 above)
Comment (vytas7) [reply tests/conftest.py:27]: No, I was thinking to take one step at a time. The comment is misleading and alluding to that the removal has happened, I will update the comment.

File: examples/recipes/output_csv_stream_asgi.py (hunk 1 above)
Comment (CaselIT) [reply examples/recipes/output_csv_stream_asgi.py:27]: interesting. that's not actually async though, but I guess it doesn't matter to the parser.

File: tests/conftest.py (hunk 3 above)
Comment (CaselIT) [reply tests/conftest.py:27]: since there are test, we can maybe to that in this pr too, but as you prefer

Comment (vytas7) [reply tests/conftest.py:27]: I'll just update the comment for now. Better keep shipping in small increments :shipit:

File: tests/test_examples.py (hunk 4 above)
Comment (vytas7) [reply tests/test_examples.py:23]: (Being discussed on Gitter)

File: tests/test_recipes.py (hunk 5 above)
Comment (vytas7) [reply tests/test_recipes.py:65]: Done, added more affordances to load module to be able to easier parametrize.

**Comment by vytas7** [comment]
> nice work!

Thanks!

**Review by CaselIT — APPROVED** [review]

**Pull request merged**